pkg/agent/controller/traceflow/packetin.go

```diff
@@ -0,0 +1,196 @@
+		tf.Status.Results = append(tf.Status.Results, *nodeResult)
+		tf, err = c.traceflowClient.OpsV1alpha1().Traceflows().Update(context.TODO(), tf, v1.UpdateOptions{})
+		if err != nil {
+			klog.Warningf("Update traceflow failed: %+v", err)
+			return err
+		}
+		klog.Infof("Updated traceflow %s: %+v", tf.Name, nodeResult)
+		return nil
+	})
+	if err != nil {
+		klog.Errorf("Update traceflow error: %+v", err)
+	}
+	return err
+}
+
+func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*opsv1alpha1.Traceflow, *opsv1alpha1.NodeResult, error) {
+	matchers := pktIn.GetMatches()
+	var match *ofctrl.MatchField
+
+	// Get data plane tag.
+	if match = getMatchRegField(matchers, uint32(openflow.TraceflowReg)); match == nil {
+		return nil, nil, errors.New("traceflow data plane tag not found")
+	}
+	rngTag := openflow13.NewNXRange(int(openflow.OfTraceflowMarkRange[0]), int(openflow.OfTraceflowMarkRange[1]))
+	tag, err := getInfoInReg(match, rngTag)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Get traceflow CRD from cache by data plane tag.
+	tf, err := c.GetRunningTraceflowCRD(uint8(tag))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	obs := make([]opsv1alpha1.Observation, 0)
+	isSender := c.isSender(uint8(tag))
+	tableID := pktIn.TableId
+
+	if isSender {
+		ob := new(opsv1alpha1.Observation)
+		ob.Component = opsv1alpha1.SpoofGuard
+		ob.Action = opsv1alpha1.Forwarded
+		obs = append(obs, *ob)
+	} else {
+		ob := new(opsv1alpha1.Observation)
+		ob.Component = opsv1alpha1.Forwarding
+		ob.Action = opsv1alpha1.Received
+		ob.ComponentInfo = openflow.GetFlowTableName(openflow.ClassifierTable)
+		obs = append(obs, *ob)
+	}
+
+	// Collect egress conjunctionID and get NetworkPolicy from cache.
+	if match = getMatchRegField(matchers, uint32(openflow.EgressReg)); match != nil {
+		egressInfo, err := getInfoInReg(match, nil)
+		if err != nil {
+			return nil, nil, err
+		}
+		ob := new(opsv1alpha1.Observation)
+		ob.Component = opsv1alpha1.NetworkPolicy
+		ob.ComponentInfo = openflow.GetFlowTableName(openflow.EgressRuleTable)
+		ob.Action = opsv1alpha1.Forwarded
+		npName, npNamespace := c.ofClient.GetPolicyFromConjunction(egressInfo)
+		if npName != "" {
+			ob.NetworkPolicy = fmt.Sprintf("%s/%s", npNamespace, npName)
+		}
+		obs = append(obs, *ob)
+	}
+
+	// Collect ingress conjunctionID and get NetworkPolicy from cache.
+	if match = getMatchRegField(matchers, uint32(openflow.IngressReg)); match != nil {
+		ingressInfo, err := getInfoInReg(match, nil)
+		if err != nil {
+			return nil, nil, err
+		}
+		ob := new(opsv1alpha1.Observation)
+		ob.Component = opsv1alpha1.NetworkPolicy
+		ob.ComponentInfo = openflow.GetFlowTableName(openflow.IngressRuleTable)
+		ob.Action = opsv1alpha1.Forwarded
+		npName, npNamespace := c.ofClient.GetPolicyFromConjunction(ingressInfo)
+		if npName != "" {
+			ob.NetworkPolicy = fmt.Sprintf("%s/%s", npNamespace, npName)
+		}
+		obs = append(obs, *ob)
+	}
+
+	// Get drop table.
+	if tableID == 60 || tableID == 100 {
```

Comment:
We should not use integer values for table IDs directly. It is hard to track any changes.